datafusion/functions/src/core/arrowtypeof.rs

```diff
@@ -69,4 +69,8 @@ impl ScalarUDFImpl for ArrowTypeOfFunc {
             "{input_data_type}"
         ))))
     }
+
+    fn validate_number_of_rows(&self) -> bool {
```

Comment:
I don't think this is correct -- arrow_typeof always returns the same number of rows as its input...

Reply:
then i think this new method is no longer needed